### ISSUE TITLE
Research how to define a collector "step" in the recipe file

### ISSUE THREAD

**Issue by joaopgrassi** [joaopgrassi/otel-recipes#195]
Today the recipe file contains a list of steps. Each step contains a field called `source` which is assumed to be always an URL to to the actual code file on GitHub.

In #194 we identified that we also need to display how to run the collector, as it's a dependency for each recipe we have today. 

In preparation for this, an initial list of things that need to change in the JSON schema are:

- Need to add a way to place "inline" code that is to be displayed on the website - e.g. to run the collector docker container
- A way to offer a direct download of the Collector config file from the recipe, so users don't have to jump back to GitHub to download the collector config.

An idea is to add a new  enum `type` field to `stepContent` that would have values `collector` and `recipe`. This way, we can add more fields that are only to be used when `type=collector`, such as collector version, run command, etc.

**Comment by joaopgrassi** [comment]
One difficulty is that the sample apps are configured to run inside compose, and talk to the collector via its dns name. If we offer a simple docker run command, the collector will run on `localhost` and the sample code from the site won't work :(.

**Comment by joaopgrassi** [comment]
There's way to make it more "generic", for ex add a step to set an env variable for the collector endpoint. and in code we refer to this env variable. 

but asking the user to define an env variable is also annoying.. I think it should be straightforward enough that we add some sort of disclaimer that when running it should be `localhost` and that's it.

**Comment by emdneto** [comment]
@joaopgrassi, what are your thoughts on using environment variables, specifically `OTEL_EXPORTER_OTLP_ENDPOINT`? One idea I had is In the recipe, we can instruct the user to set `OTEL_EXPORTER_OTLP_ENDPOINT="http://localhost:4317"` and then run the docker run command. In a docker-compose setup, we can configure the environment variable for the app to communicate with the collector through DNS. This approach would allow users to easily copy and paste the recipe while keeping our tests operational. However, if this turns out to be challenging to implement, I believe that your solution is enough, at least for the time being.

**Comment by joaopgrassi** [comment]
Yeah I was also thinking about it, I don't love the idea as it adds another step but I think it's the best alternative. I'm already experimenting with it, will be opening a PR soon. I'm currently struggling with the website and css things :D
